unstructured/partition/lang.py

```diff
@@ -167,34 +167,68 @@ def prepare_languages_for_tesseract(languages: Optional[List[str]] = ["eng"]):
     return TESSERACT_LANGUAGES_SPLITTER.join(converted_languages)
 
 
-def check_languages(languages: Optional[List[str]], ocr_languages: Optional[str]):
-    """Handle `ocr_languages` and `languages`, defining `languages` to ['eng'] as default and
-    converting `ocr_languages` if needed"""
-    if languages is None:
-        languages = ["eng"]
+def check_language_args(
+    languages: Optional[List[str]], ocr_languages: Optional[str]
+) -> Union[list[str], None]:
```

Comment:
The return type should be `Optional[list[str]]`. If you were going to use `Union` notation for this, which is perfectly valid, just not conventional for this code base, then it would be `list[str] | None`. Both the lowercase version of `List` and the `|` version of `Union` require a `from __future__ import annotations` import (which all our files should have until Python 3.13 or later).

Reply:
I would be strongly inclined to cut off optionality of `languages` at the interface function and use `[]` as the zero value for `languages` everywhere else. Rationale:

1. The option to provide None is a convenience to an end-user who is specifying `languages` as a kwarg (dict item) where it it more work to omit the argument. If a default is applied to replace an omitted value, it's the interface function's job to do that substitution.
2. `None` and `[]` have the same semantics (as far as I know). Using both complicates any other non-interface function that uses that value. In general, `Optional` should be used sparingly outside of interface functions and methods.

The general principle here is to identify flexibilities we afford an end-user in calling an interface function and isolate those flexibilities to that interface function. Otherwise we have internal and helper functions attending to peculiarities of calling flexibility which should not be their responsibility (and makes interface changes like this tend to involve much deeper parts of the code than we'd like because they're unnecessarily coupled to the interface particulars.